libqmi: 1.28.8 -> 1.30.2, modemmanager: 1.16.10 -> 1.18.2

amaxine opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
https://lists.freedesktop.org/archives/modemmanager-devel/2021-September/008826.html

It would be good to add support for passing the `--test-quick-suspend-resume` to the daemon, but we currently don't have any configuration available for modemmanager and I'm not super into dealing with it right now :)

I'm not switching the build to meson yet, quoting from the announcement "the meson port is not fully complete yet". Probably in the next version.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Commit:
libqmi: 1.28.8 -> 1.30.2

- https://gitlab.freedesktop.org/mobile-broadband/libqmi/-/blob/1.30.2/NEWS
- https://lists.freedesktop.org/archives/libqmi-devel/2021-August/003667.html
- https://lists.freedesktop.org/archives/libqmi-devel/2021-September/003697.html

Commit:
modemmanager: 1.16.10 -> 1.18.2

- https://lists.freedesktop.org/archives/modemmanager-devel/2021-September/008826.html
- https://lists.freedesktop.org/archives/modemmanager-devel/2021-September/008836.html

jtojnar commented:
Do we still need `--filter-policy` in `nixos/modules/services/networking/networkmanager.nix`?

maxeaubrey commented:
Upstream has made some changes (adding capabilities to the unit file), but as far as I can tell that setting isn't default yet?

Commit:
nixos/networkmanager: drop modemmanager overrides

as of 1.18.0, modemmanager defaults to strict when no other options are
passed, so we can just use the upstream unit as is

maxeaubrey commented:
Sorry, you're right, strict is now default when filter policy is unset. I've fixed that with the new bump.

jtojnar commented on nixos/modules/services/networking/networkmanager.nix:
```diff
@@ -505,7 +505,7 @@ in {
     # TODO: keep an eye on modem-manager releases as this will eventually be added to the upstream unit
     systemd.services.ModemManager.serviceConfig.ExecStart = [
       ""
-      "${pkgs.modemmanager}/sbin/ModemManager --filter-policy=STRICT"
```
Actually, we should just get rid of this option definition altogether.

maxeaubrey replied:
Oh yeah, no point in overriding things anymore. I'll fix that up.

tomfitzhenry commented:
This will be land in master as of https://github.com/NixOS/nixpkgs/pull/141684. https://nixpk.gs/pr-tracker.html?pr=137233
